networkx/algorithms/community/modularity_max.py

```diff
@@ -149,9 +132,7 @@ def greedy_modularity_communities(G, weight=None, resolution=1, n_communities=1)
         else:
             # Duplicate wasn't in H, just remove from row v heap
             dq_heap[v].remove((v, u))
-        # Stop when change is non-positive (no improvement possible)
-        if dq <= 0:
-            break
+        yield dq
```

Comment:
Minor detail, but shouldn't we just yield this after line 117?  Messing with `dq_heap` isn't needed by the calling function for this `yield`.